### PR TITLE
add chain id

### PR DESCRIPTION
**Pull request by Thykof**
related to https://github.com/massalabs/massa/issues/4583

## Delivery Quality Checklist

- [ ] **Breaking Changes in API:**
  Does this PR introduce breaking changes in the API?
    - If yes, have you considered making it backward compatible?
    - If backward compatibility is not considered, set the "breaking-change" label.

- [ ] **Changelog:**
  - [ ] For bugfix PR, set the "bugfix" label
  - [ ] If this change should not appear in changelog, use "ignore-for-changelog" label

- [ ] **Version Update Handling:**
  Have you ensured that the version update by user is handled correctly?

- [ ] **PR Dependency:**
  Does this PR depend on another PR?
    - If yes, is it necessary for the dependency to be released prior to merging this one?

- [ ] **Documentation:**
  - [ ] Are any necessary changes made to user-facing documentation?
  - [ ] Confirm that API documentation is updated with any relevant changes.
  - [ ] Check that README and other documentation files are accurate and current.

### CODE REVIEW

File: int/config/network.go [hunk 1]
```diff
@@ -166,16 +135,33 @@ func (n *NetworkManager) SwitchNetwork(selectedNetworkStr string) error {
...
+
+	version, err := versionFromStatus(status)
 	if err != nil {
 		return fmt.Errorf("getting network version: %w", err)
 	}
 
+	// compare chain id from node status with chain id from config
+	nodeChainID := uint64(*status.ChainID)
+	if nodeChainID != cfg.ChainID {
+		logger.Errorf("chain id mismatch: %d != %d", nodeChainID, cfg.ChainID)
```
Comment (Thykof) [inline int/config/network.go:154]: should we exit here?

```suggestion
		logger.Fatalf("chain id mismatch: %d != %d", nodeChainID, cfg.ChainID)
```

Comment (thomas-senechal) [reply int/config/network.go:154]: I guess we shouldn't exit. It's a node issue. 
But we could display an error (or a warning) to tell user that there is something weird with the connected node and so, that we recommand switching node.

Maybe that should be done in another issue/PR as I don't think it is a priority

**Review by thomas-senechal — APPROVED** [review]

**Comment by damip** [comment]
CI seems to fail

**Comment by Thykof** [comment]
> CI seems to fail

CI fails because buildnet is not handling chain id yet

**Review by peterjah — APPROVED** [review]

**Pull request merged**